docs/architecture/Overview.md

```diff
@@ -0,0 +1,164 @@
+
+### Stateful clients
+
+Directories:
+- [`calling-stateful-client`](../../packages/calling-stateful-client/)
+- [`chat-statefule-client`](../../packages/chat-stateful-client/)
+
+Ultimately, this library depends on the headless SDKs to interact with ACS backend services ([calling](https://www.npmjs.com/package/@azure/communication-calling) and [chat](https://www.npmjs.com/package/@azure/communication-chat)).
+
+- The stateful clients are thin wrappers that are API compatible with the primary clients from the headless packages.
+- The primary contribution of the stateful clients is that they cache all state updates from the backend and expose the entire backend state as a single rooted immutable directed acyclic graph. This way of exposing cached state allows the UI to use references to nodes in the state graph as proxies for the structure of data under the nodes.
+- The stateful wrappers also support better error handling by [bubbling up errors to the application in a variety of ways](./ErrorReporting.md).
+
+### UI components
+
+Directories:
+- [`react-components`](../../packages/react-components/)
+
+The other end of the component layer is a set of highly customizable (mostly) pure UI components. These UI components are agnostic of ACS concepts (and in particular the stateful clients) and are entirely driven by data and callback props defined within this packlet. See [this related doc](./CustomizableComponent.md) for some inspiration on how to design good UI components.
+
+### Bindings
+
+The two parts of the component layer - stateful clients and UI components define their own API. The binding layer provides the bridge that connects these two.
+
+- [Selectors](./WritingSelectors.md) effeciently bind UI component data props to specific nodes in the immutable state exposed by the stateful layer. Selectors are setup to (1) avoid unnecessary component rerenders, and (2) avoid costly computation to determine when the component needs to rerender.
+- Handlers bind UI component callback props to methods exposed by the stateful layer.
+
+For both selectors and handlers, the library exposes a very simple API for customers wanting to connect the UI components and the stateful clients without any customization - a single `usePropsFor` hook. The library also exposes a more verbose API for sophisticated use-cases. This progressive disclosure of detail allows customers to do simple things easily without limiting more complex scenarios.
```

Comment:
Not sure if we want to mention useSelector here as well - it's also a public exposed api, but for more advanced developers

Reply:
Will skip. The intention is not to be exhaustive in this doc.